libr/bin/format/dex/dex.c

```diff
@@ -25,14 +26,32 @@ struct r_bin_dex_obj_t* r_bin_dex_new_buf(RBuffer *buf) {
 	if (!r_buf_set_bytes (bin->b, buf->buf, bin->size)) {
 		goto fail;
 	}
-	// XXX: this is not endian safe
-	// XXX: no need to dup all data!! just pointers to the bin->b
-	// XXX: return value not checked
 	/* header */
-	//r_buf_read_at (bin->b, 0, (ut8*)&bin->header, sizeof (struct dex_header_t));
 	if (bin->size < sizeof(struct dex_header_t))
 		goto fail;
-	bin->header = (*(struct dex_header_t*)bin->b->buf);
+	memcpy (&bin->header.magic, bin->b->buf, 8);
+	bin->header.checksum = r_read_le32 (bin->b->buf + 8);
```

Comment:
add a new pointer that just points to bin->b->buf. there's no need to dereference two pointers on every single line